dataprofiler/profilers/column_profile_compilers.py

```diff
@@ -325,9 +325,36 @@ def profile(self):
         }
         # TODO: Only works for last profiler. Abstracted for now.
         for _, profiler in self._profiles.items():
-            profile["data_label"] = profiler.data_label
-            profile["statistics"].update(profiler.profile)
+            col_profile = profiler.profile
+            profile["data_label"] = col_profile.pop("data_label")
+            profile["statistics"].update(col_profile)
         return profile
+    
+    def diff(self, other, options=None):
+        """
+        Finds the difference between 2 compilers and returns the report
+
+        :param other: profile compiler finding the difference with this one.
+        :type other: ColumnDataLabelerCompiler
+        :return: difference of the profiles
+        :rtype: dict
+        """
+        # Call super for compiler instance check
+        diff_profile = super().diff(other, options)
+        diff_profile["statistics"] = dict()
+
+        # Iterate through profile(s)
+        all_profiles = set(self._profiles.keys()) | set(other._profiles.keys())
```

Comment:
can we just do `all_profiles = set(self._profiles.keys()) & set(other._profiles.keys())`, then we don't need the if statement down below inside the for loop?